src/irr/asset/CPLYMeshFileLoader.cpp

```diff
@@ -261,41 +261,42 @@ asset::SAssetBundle CPLYMeshFileLoader::loadAsset(io::IReadFile* _file, const as
 				}
 			}
 
+			asset::SBufferBinding<ICPUBuffer> bufferBinding;
+
             if (indices.size())
             {
-				ICPUMeshBuffer::SBufferBinding bufferBinding;
 				bufferBinding.buffer = core::make_smart_refctd_ptr<asset::ICPUBuffer>(indices.size() * sizeof(uint32_t));
+				bufferBinding.offset = 0ull;
                 memcpy(bufferBinding.buffer->getPointer(), indices.data(), bufferBinding.buffer->getSize());
 
-				if (!genVertBuffersForMBuffer(mb.get(), attribs, bufferBinding))
-					return {};
+				mb->setIndexBufferBinding(std::move(bufferBinding));
 
                 mb->setIndexCount(indices.size());
                 mb->setIndexType(asset::EIT_32BIT);
 				mb->getPipeline()->getPrimitiveAssemblyParams().primitiveType = E_PRIMITIVE_TOPOLOGY::EPT_TRIANGLE_LIST;
+
+				if (!genVertBuffersForMBuffer(mb.get(), attribs, bufferBinding))
+					return {};
             }
             else
             {
-				mb->getPipeline()->getPrimitiveAssemblyParams().primitiveType = E_PRIMITIVE_TOPOLOGY::EPT_POINT_LIST;
-                mb->setIndexCount(attribs[E_POS].size());
-                //mb->getMaterial().setFlag(video::EMF_POINTCLOUD, true);
+				mb->getPipeline()->getPrimitiveAssemblyParams().primitiveType = E_PRIMITIVE_TOPOLOGY::EPT_TRIANGLE_FAN;
+				mb->setIndexCount(0);
```

Comment:
index count is the number of vertices that will be used, CANNOT BE 0

Reply:
our documentation after peeking definition of functions says: 

> //! Sets amount of indices.
> 	/** @returns Whether set amount exceeds mapped buffer's size. Regardless of result the amount is set. */

and in that section there is no indices, so I state it explicitly writing `0`